Sming/Arch/Host/Components/spi_flash/flashmem.cpp

```diff
@@ -160,6 +179,10 @@ uint32_t flashmem_write(const void* from, uint32_t toaddr, uint32_t size)
 
 uint32_t flashmem_read(void* to, uint32_t fromaddr, uint32_t size)
 {
+	if(readRealMem(to, fromaddr, size)) {
```

Comment:
Just for my understanding: 

Let's say a developer has the following application that she/he wants to test on the host emulator:

1. App with two ROMs.
2. No matter which ROM is selected the app should be able to read data stored in the flash starting from the 3rd MB.
3. No matter which ROM is selected the app should be able to write data to the flash at an address starting from the 3rd MB.

That was working correctly up to now. Will it continue to work with this PR?

Reply:
Yes, it'll work exactly as before. Behaviour only changes for reads above 2GB so we're fine below that.

Both Linux and Windows reserve addresses above 2GB for system use so we'll never see them passed as pointers from applications. All that's happening is we're mapping the application's *real* pointer addresses into this unused space.